### PR TITLE
Improve replay playback accuracy

### PR DESCRIPTION
**Pull request by peppy**
Ensure important frames are accounted for, as well as potential client lag. This is open as a RFC and will likely see a few more commits before it is in a happy state.

- [x] Depends on https://github.com/ppy/osu-framework/pull/663 for `UpdateSubTree` accessibility.
- [x] Depends on https://github.com/ppy/osu-framework/pull/667

### CODE REVIEW

**Comment by peppy** [comment]
Note that I can still trigger scenarios where this is not frame-accurate. May be spinner related (requires further testing) but this can be done in future work.

**Comment by peppy** [comment]
Note that CI won't pass until the framework-level dependency is merged.

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -165,16 +180,20 @@ private void load(AudioManager audio, BeatmapDatabase beatmaps, OsuConfigManager
...
+                            RelativeSizeAxes = Axes.Both,
+                            Clock = offsetClock,
+                            Children = new Drawable[]
+                            {
+                                HitRenderer,
+                                skipButton = new SkipButton { Alpha = 0 },
+                            }
                         },
                     }
                 },
```
Comment (smoogipooo) [inline osu.Game/Screens/Play/Player.cs]: Is there a reason we have two nested `Containers` here? Can't this be `hitRendererContainer { Clock = offsetClock, Children = { hitRenderer, skipButton } }` ?

Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: so a fade can be applied in game clock time, not audio clock time.

Comment (smoogipooo) [inline osu.Game/Screens/Play/Player.cs]: ok

**Review by smoogipoo — DISMISSED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**